### PR TITLE
Provide str() and repr() representations for sections and segments

### PR DESCRIPTION
**Pull request by zachriggle**

### CODE REVIEW

File: elftools/elf/segments.py [hunk 1]
```diff
@@ -79,6 +79,14 @@ def section_in_segment(self, section):
                 secoffset - poffset + section['sh_size'] <= self['p_filesz'] and
                 secoffset - poffset <= self['p_filesz'] - 1)
 
+    def __str__(self):
+        try: return self.header.p_type
+        except: return super(Segment, self).__str__()
```
Comment (vadmium) [inline elftools/elf/segments.py]: You don’t want to be handling asynchronous exceptions like KeyboardInterrupt. Change to “except Exception”, or maybe you’re only actually interested in AttributeError or something?

**Comment by eliben** [comment]
Please add tests and squash to a single commit

**Comment by LeonardMH** [comment]
@eliben Same question here, what needs to be done to accept this PR? I'd like to see it added and can possibly work on it in the upcoming weeks.

**Comment by eliben** [comment]
@LeonardMH my reply to the PR summarizes it, I think. Tests + squash in a single commit. You can create a separate PR if you want - this one has long been inactive

**Comment by eliben** [comment]
Closing due to inactivity. If anyone is willing to reopen and rebase this, please open a new PR

**Pull request closed without merging**